packages/frontend-shared/src/utils/isRunMode.ts

```diff
@@ -1 +1,3 @@
-export const isRunMode = window.__CYPRESS_MODE__ === 'run' && window.top === window
+import { CY_IN_CY_SIMULATE_RUN_MODE } from '@packages/types/src/constants'
+
+export const isRunMode = window.location.href.includes(CY_IN_CY_SIMULATE_RUN_MODE) || (window.__CYPRESS_MODE__ === 'run' && window.top === window)
```

Comment:
Could this be another window variable? We could set something like `window.__CYPRESS_CY_IN_CY_SIMULATE_RUN_MODE`. I think the implementation of this would be a bit more involved since you might have to funnel this variable into the serving of the app index.html but the benefit would be that it would fit into the existing strategy. With this we now have state stored on the window and the URL

Reply:
Initially it was on the window and like you picked up on, Cypress setting it is too late in the process, so I think you're right, we'd have to modify the page itself before it's served, or hook in earlier so that the page renders in run mode.

I think it's OK to use the URL to tell a page to start out in a particular state. It's also kinda neat if you are in actual open mode and want to flip to "simulated run mode".

Opening to revisiting at any time.

Reply:
The toggling between seems cool! Good way to test the run mode vs open mode UI.